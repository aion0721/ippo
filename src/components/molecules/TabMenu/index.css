.tabs-fill .bp3-tab-indicator-wrapper {
  display: inline-block;
}

.tabs-fill .bp3-tab-indicator-wrapper:nth-child(1) {
  width: 100%;
}

.tabs-fill .bp3-tab-indicator-wrapper:nth-child(2) {
  width: 50%;
}

.tabs-fill .bp3-tab-indicator-wrapper:nth-child(3) {
  width: 33%;
}

.tabs-fill .bp3-tab-indicator-wrapper:nth-child(4) {
  width: 25%;
}

.tabs-fill .bp3-tab {
  display: inline-block;
  text-align: center;
  margin-right: 0;
}

.tabs-fill .bp3-tab:nth-last-child(1) {
  width: 100%;
}

.tabs-fill .bp3-tab:nth-last-child(2),
.tabs-fill .bp3-tab:nth-last-child(2) ~ .bp3-tab {
  width: 50%;
}

.tabs-fill .bp3-tab:nth-last-child(3),
.tabs-fill .bp3-tab:nth-last-child(3) ~ .bp3-tab {
  width: 33%;
}

.tabs-fill .bp3-tab:nth-last-child(4),
.tabs-fill .bp3-tab:nth-last-child(4) ~ .bp3-tab {
  width: 25%;
}