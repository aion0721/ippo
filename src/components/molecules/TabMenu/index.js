import React from 'react';
import { Tab, Tabs } from '@blueprintjs/core';
import propTypes from 'prop-types';
import './index.css';

const TabMenu = ({ items }) => (
  <Tabs className="tabs-fill">
    {items.map(({ Component, id, title }) => (
      <Tab key={id} id={id} title={title} panel={<Component />} />
    ))}
  </Tabs>
);

TabMenu.displayName = 'TabMenu';

TabMenu.propTypes = {
  items: propTypes.arrayOf(
    propTypes.shape({
      id: propTypes.string,
      title: propTypes.string,
      Component: propTypes.element,
    }),
  ),
};

TabMenu.defaultProps = {
  items: [],
};

export default TabMenu;
