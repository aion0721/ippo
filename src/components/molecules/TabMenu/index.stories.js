import React from 'react';
import { storiesOf } from '@storybook/react';
import TabMenu from '.';

const stories = storiesOf('molecules/TabMenu', module);

const item = i => ({
  id: String(i),
  title: String(i),
  Component: () => <p>{String(i)}</p>,
});

const props = times => ({
  items: [...Array(times)].map((_, i) => item(i)),
});

stories.add('タブが1の場合', () => <TabMenu {...props(1)} />);

stories.add('タブが2の場合', () => <TabMenu {...props(2)} />);

stories.add('タブが3つ場合', () => <TabMenu {...props(3)} />);

stories.add('タブが4つ場合', () => <TabMenu {...props(4)} />);
