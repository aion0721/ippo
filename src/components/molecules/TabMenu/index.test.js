import React from 'react';
import { snapshot } from 'test/helpers';
import TabMenu from '.';

const item = i => ({
  id: String(i),
  title: String(i),
  Component: () => <p>{String(i)}</p>,
});

const props = times => ({
  items: [...Array(times)].map((_, i) => item(i)),
});

snapshot('TabMenu/tab1', <TabMenu {...props(1)} />);

snapshot('TabMenu/tab2', <TabMenu {...props(2)} />);

snapshot('TabMenu/tab3', <TabMenu {...props(3)} />);

snapshot('TabMenu/tab4', <TabMenu {...props(4)} />);
