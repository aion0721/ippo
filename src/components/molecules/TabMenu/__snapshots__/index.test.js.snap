// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: TabMenu/tab1 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <TabMenu
    items={
      Array [
        Object {
          "Component": [Function],
          "id": "0",
          "title": "0",
        },
      ]
    }
  >
    <Blueprint3.Tabs
      animate={true}
      className="tabs-fill"
      large={false}
      renderActiveTabPanelOnly={false}
      vertical={false}
    >
      <div
        className="bp3-tabs tabs-fill"
      >
        <div
          className="bp3-tab-list"
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          role="tablist"
        >
          <div
            className="bp3-tab-indicator-wrapper"
            style={
              Object {
                "height": 0,
                "transform": "translateX(0px) translateY(0px)",
                "width": 0,
              }
            }
          >
            <div
              className="bp3-tab-indicator"
            />
          </div>
          <Blueprint3.TabTitle
            disabled={false}
            id="0"
            key=".$0"
            onClick={[Function]}
            panel={<Component />}
            selected={true}
            title="0"
          >
            <div
              aria-controls="bp3-tab-panel_undefined_0"
              aria-disabled={false}
              aria-expanded={true}
              aria-selected={true}
              className="bp3-tab"
              data-tab-id="0"
              id="bp3-tab-title_undefined_0"
              onClick={[Function]}
              role="tab"
              tabIndex={0}
            >
              0
            </div>
          </Blueprint3.TabTitle>
        </div>
        <div
          aria-hidden={false}
          aria-labelledby="bp3-tab-title_undefined_0"
          className="bp3-tab-panel"
          id="bp3-tab-panel_undefined_0"
          key="0"
          role="tabpanel"
        >
          <Component>
            <p>
              0
            </p>
          </Component>
        </div>
      </div>
    </Blueprint3.Tabs>
  </TabMenu>
</MockRouter>
`;

exports[`should match snapshot: TabMenu/tab2 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <TabMenu
    items={
      Array [
        Object {
          "Component": [Function],
          "id": "0",
          "title": "0",
        },
        Object {
          "Component": [Function],
          "id": "1",
          "title": "1",
        },
      ]
    }
  >
    <Blueprint3.Tabs
      animate={true}
      className="tabs-fill"
      large={false}
      renderActiveTabPanelOnly={false}
      vertical={false}
    >
      <div
        className="bp3-tabs tabs-fill"
      >
        <div
          className="bp3-tab-list"
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          role="tablist"
        >
          <div
            className="bp3-tab-indicator-wrapper"
            style={
              Object {
                "height": 0,
                "transform": "translateX(0px) translateY(0px)",
                "width": 0,
              }
            }
          >
            <div
              className="bp3-tab-indicator"
            />
          </div>
          <Blueprint3.TabTitle
            disabled={false}
            id="0"
            key=".$0"
            onClick={[Function]}
            panel={<Component />}
            selected={true}
            title="0"
          >
            <div
              aria-controls="bp3-tab-panel_undefined_0"
              aria-disabled={false}
              aria-expanded={true}
              aria-selected={true}
              className="bp3-tab"
              data-tab-id="0"
              id="bp3-tab-title_undefined_0"
              onClick={[Function]}
              role="tab"
              tabIndex={0}
            >
              0
            </div>
          </Blueprint3.TabTitle>
          <Blueprint3.TabTitle
            disabled={false}
            id="1"
            key=".$1"
            onClick={[Function]}
            panel={<Component />}
            selected={false}
            title="1"
          >
            <div
              aria-controls="bp3-tab-panel_undefined_1"
              aria-disabled={false}
              aria-expanded={false}
              aria-selected={false}
              className="bp3-tab"
              data-tab-id="1"
              id="bp3-tab-title_undefined_1"
              onClick={[Function]}
              role="tab"
              tabIndex={0}
            >
              1
            </div>
          </Blueprint3.TabTitle>
        </div>
        <div
          aria-hidden={false}
          aria-labelledby="bp3-tab-title_undefined_0"
          className="bp3-tab-panel"
          id="bp3-tab-panel_undefined_0"
          key="0"
          role="tabpanel"
        >
          <Component>
            <p>
              0
            </p>
          </Component>
        </div>
        <div
          aria-hidden={true}
          aria-labelledby="bp3-tab-title_undefined_1"
          className="bp3-tab-panel"
          id="bp3-tab-panel_undefined_1"
          key="1"
          role="tabpanel"
        >
          <Component>
            <p>
              1
            </p>
          </Component>
        </div>
      </div>
    </Blueprint3.Tabs>
  </TabMenu>
</MockRouter>
`;

exports[`should match snapshot: TabMenu/tab3 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <TabMenu
    items={
      Array [
        Object {
          "Component": [Function],
          "id": "0",
          "title": "0",
        },
        Object {
          "Component": [Function],
          "id": "1",
          "title": "1",
        },
        Object {
          "Component": [Function],
          "id": "2",
          "title": "2",
        },
      ]
    }
  >
    <Blueprint3.Tabs
      animate={true}
      className="tabs-fill"
      large={false}
      renderActiveTabPanelOnly={false}
      vertical={false}
    >
      <div
        className="bp3-tabs tabs-fill"
      >
        <div
          className="bp3-tab-list"
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          role="tablist"
        >
          <div
            className="bp3-tab-indicator-wrapper"
            style={
              Object {
                "height": 0,
                "transform": "translateX(0px) translateY(0px)",
                "width": 0,
              }
            }
          >
            <div
              className="bp3-tab-indicator"
            />
          </div>
          <Blueprint3.TabTitle
            disabled={false}
            id="0"
            key=".$0"
            onClick={[Function]}
            panel={<Component />}
            selected={true}
            title="0"
          >
            <div
              aria-controls="bp3-tab-panel_undefined_0"
              aria-disabled={false}
              aria-expanded={true}
              aria-selected={true}
              className="bp3-tab"
              data-tab-id="0"
              id="bp3-tab-title_undefined_0"
              onClick={[Function]}
              role="tab"
              tabIndex={0}
            >
              0
            </div>
          </Blueprint3.TabTitle>
          <Blueprint3.TabTitle
            disabled={false}
            id="1"
            key=".$1"
            onClick={[Function]}
            panel={<Component />}
            selected={false}
            title="1"
          >
            <div
              aria-controls="bp3-tab-panel_undefined_1"
              aria-disabled={false}
              aria-expanded={false}
              aria-selected={false}
              className="bp3-tab"
              data-tab-id="1"
              id="bp3-tab-title_undefined_1"
              onClick={[Function]}
              role="tab"
              tabIndex={0}
            >
              1
            </div>
          </Blueprint3.TabTitle>
          <Blueprint3.TabTitle
            disabled={false}
            id="2"
            key=".$2"
            onClick={[Function]}
            panel={<Component />}
            selected={false}
            title="2"
          >
            <div
              aria-controls="bp3-tab-panel_undefined_2"
              aria-disabled={false}
              aria-expanded={false}
              aria-selected={false}
              className="bp3-tab"
              data-tab-id="2"
              id="bp3-tab-title_undefined_2"
              onClick={[Function]}
              role="tab"
              tabIndex={0}
            >
              2
            </div>
          </Blueprint3.TabTitle>
        </div>
        <div
          aria-hidden={false}
          aria-labelledby="bp3-tab-title_undefined_0"
          className="bp3-tab-panel"
          id="bp3-tab-panel_undefined_0"
          key="0"
          role="tabpanel"
        >
          <Component>
            <p>
              0
            </p>
          </Component>
        </div>
        <div
          aria-hidden={true}
          aria-labelledby="bp3-tab-title_undefined_1"
          className="bp3-tab-panel"
          id="bp3-tab-panel_undefined_1"
          key="1"
          role="tabpanel"
        >
          <Component>
            <p>
              1
            </p>
          </Component>
        </div>
        <div
          aria-hidden={true}
          aria-labelledby="bp3-tab-title_undefined_2"
          className="bp3-tab-panel"
          id="bp3-tab-panel_undefined_2"
          key="2"
          role="tabpanel"
        >
          <Component>
            <p>
              2
            </p>
          </Component>
        </div>
      </div>
    </Blueprint3.Tabs>
  </TabMenu>
</MockRouter>
`;

exports[`should match snapshot: TabMenu/tab4 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <TabMenu
    items={
      Array [
        Object {
          "Component": [Function],
          "id": "0",
          "title": "0",
        },
        Object {
          "Component": [Function],
          "id": "1",
          "title": "1",
        },
        Object {
          "Component": [Function],
          "id": "2",
          "title": "2",
        },
        Object {
          "Component": [Function],
          "id": "3",
          "title": "3",
        },
      ]
    }
  >
    <Blueprint3.Tabs
      animate={true}
      className="tabs-fill"
      large={false}
      renderActiveTabPanelOnly={false}
      vertical={false}
    >
      <div
        className="bp3-tabs tabs-fill"
      >
        <div
          className="bp3-tab-list"
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          role="tablist"
        >
          <div
            className="bp3-tab-indicator-wrapper"
            style={
              Object {
                "height": 0,
                "transform": "translateX(0px) translateY(0px)",
                "width": 0,
              }
            }
          >
            <div
              className="bp3-tab-indicator"
            />
          </div>
          <Blueprint3.TabTitle
            disabled={false}
            id="0"
            key=".$0"
            onClick={[Function]}
            panel={<Component />}
            selected={true}
            title="0"
          >
            <div
              aria-controls="bp3-tab-panel_undefined_0"
              aria-disabled={false}
              aria-expanded={true}
              aria-selected={true}
              className="bp3-tab"
              data-tab-id="0"
              id="bp3-tab-title_undefined_0"
              onClick={[Function]}
              role="tab"
              tabIndex={0}
            >
              0
            </div>
          </Blueprint3.TabTitle>
          <Blueprint3.TabTitle
            disabled={false}
            id="1"
            key=".$1"
            onClick={[Function]}
            panel={<Component />}
            selected={false}
            title="1"
          >
            <div
              aria-controls="bp3-tab-panel_undefined_1"
              aria-disabled={false}
              aria-expanded={false}
              aria-selected={false}
              className="bp3-tab"
              data-tab-id="1"
              id="bp3-tab-title_undefined_1"
              onClick={[Function]}
              role="tab"
              tabIndex={0}
            >
              1
            </div>
          </Blueprint3.TabTitle>
          <Blueprint3.TabTitle
            disabled={false}
            id="2"
            key=".$2"
            onClick={[Function]}
            panel={<Component />}
            selected={false}
            title="2"
          >
            <div
              aria-controls="bp3-tab-panel_undefined_2"
              aria-disabled={false}
              aria-expanded={false}
              aria-selected={false}
              className="bp3-tab"
              data-tab-id="2"
              id="bp3-tab-title_undefined_2"
              onClick={[Function]}
              role="tab"
              tabIndex={0}
            >
              2
            </div>
          </Blueprint3.TabTitle>
          <Blueprint3.TabTitle
            disabled={false}
            id="3"
            key=".$3"
            onClick={[Function]}
            panel={<Component />}
            selected={false}
            title="3"
          >
            <div
              aria-controls="bp3-tab-panel_undefined_3"
              aria-disabled={false}
              aria-expanded={false}
              aria-selected={false}
              className="bp3-tab"
              data-tab-id="3"
              id="bp3-tab-title_undefined_3"
              onClick={[Function]}
              role="tab"
              tabIndex={0}
            >
              3
            </div>
          </Blueprint3.TabTitle>
        </div>
        <div
          aria-hidden={false}
          aria-labelledby="bp3-tab-title_undefined_0"
          className="bp3-tab-panel"
          id="bp3-tab-panel_undefined_0"
          key="0"
          role="tabpanel"
        >
          <Component>
            <p>
              0
            </p>
          </Component>
        </div>
        <div
          aria-hidden={true}
          aria-labelledby="bp3-tab-title_undefined_1"
          className="bp3-tab-panel"
          id="bp3-tab-panel_undefined_1"
          key="1"
          role="tabpanel"
        >
          <Component>
            <p>
              1
            </p>
          </Component>
        </div>
        <div
          aria-hidden={true}
          aria-labelledby="bp3-tab-title_undefined_2"
          className="bp3-tab-panel"
          id="bp3-tab-panel_undefined_2"
          key="2"
          role="tabpanel"
        >
          <Component>
            <p>
              2
            </p>
          </Component>
        </div>
        <div
          aria-hidden={true}
          aria-labelledby="bp3-tab-title_undefined_3"
          className="bp3-tab-panel"
          id="bp3-tab-panel_undefined_3"
          key="3"
          role="tabpanel"
        >
          <Component>
            <p>
              3
            </p>
          </Component>
        </div>
      </div>
    </Blueprint3.Tabs>
  </TabMenu>
</MockRouter>
`;
